packages/core/src/internal/reconciliation/futures/reconcileNamedContractDeployment.ts

```diff
@@ -43,6 +44,11 @@ export async function reconcileNamedContractDeployment(
     return result;
   }
 
+  result = reconcileStrategy(future, executionState, context);
```

Comment:
We should only reconcile the strategy if the execution hasn't completed. i.e. if a future has already been successfully executed, we don't care about its strategy.